lib/read-from-url.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const tmp = require('tmp');
+const fetch = require('minipass-fetch');
+const fs = require('fs');
+const debug = require('debug')('supported:read-from-url');
+
+const RAW_CONTENT_HOST = 'raw.githubusercontent.com';
+
+async function setupProjectPath(_url, rawFileHost) {
+  let tempDir = tmp.dirSync();
+
+  let url = new URL(_url);
```

Comment:
How do we authenticate to private github instances?

Reply:
For the private repository, I found a solution.
`https://$TOKEN.raw.test.githubprivate.com/org/repo/master/package.json` will give us the file.

Users need to pass in the token when they provide a private instance. if not, the error is thrown at the terminal to pass in the token and how to generate the token. 
Ex:
```sh
$ bin/supported https://test.githubprivate.com/stefanpenner/supported
Private instances of github needs token. To generate token follow https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
Missing required flag
	--token, -t
```